### PR TITLE
Added in the functionality to make the dismissable leaderboard work.

### PR DESCRIPTION
**Pull request by mhwatson**

### CODE REVIEW

File: client/src/containers/leaderboard.js [hunk 1]
```diff
@@ -45,93 +45,75 @@ const styles = {
   },
 };
 
-class DoggoCard extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { clicked: false };
+const DoggoCard = (props) => {
```
Comment (mhwatson) [inline client/src/containers/leaderboard.js]: Made the Card Functional. Trust, you don't want state at the lowest level unless it's like a stateful button.

File: client/src/containers/leaderboard.js [hunk 2]
```diff
@@ -45,93 +45,75 @@ const styles = {
...
+      <main>
+        <Grid
+          container
+          direction="row"
+          justify="space-evenly"
+          alignItems="center">
+          <div style={styles.card}>
+            {cardList.map(({ name, clicked }, i) => {
+              return (
+              <Slide in={!clicked} direction={this.props.pos} timeout={{enter:0, exit:250}} >
```
Comment (mhwatson) [inline client/src/containers/leaderboard.js]: Moved this to the parent level, so even if the card exits before the transition,  it looks normal.

File: client/src/containers/leaderboard.js [hunk 3]
```diff
@@ -45,93 +45,75 @@ const styles = {
...
-      </Slide>
-    </div>
+      <main>
+        <Grid
+          container
+          direction="row"
+          justify="space-evenly"
+          alignItems="center">
+          <div style={styles.card}>
+            {cardList.map(({ name, clicked }, i) => {
```
Comment (mhwatson) [inline client/src/containers/leaderboard.js]: Theres like 5 ways of doing this, and this is probably one of the ugliest ways of doing it, but it works for now.

File: client/src/containers/leaderboard.js [hunk 4]
```diff
@@ -45,93 +45,75 @@ const styles = {
...
 
-    this.onClick = () => {
-      this.setState({ clicked: true });
-      console.log('Clicked', this.state.clicked);
-    };
+class Leaderboard extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+       cardList: [
```
Comment (mhwatson) [inline client/src/containers/leaderboard.js]: Added IDs to make it clear. Weird fact: The last number is rendered on top!

File: client/src/containers/leaderboard.js [hunk 5]
```diff
@@ -45,93 +45,75 @@ const styles = {
...
+          style={styles.media} />
+        <CardContent>
+          <Typography gutterBottom variant="h5" component="h2">
+            {`Cute Doggo #${name}`}
+          </Typography>
+          <Typography component="p">
+            {"Isn't he a good boy?"}
+          </Typography>
+        </CardContent>
+        <Button size="small" color="primary" onClick={onClick} >
```
Comment (mhwatson) [inline client/src/containers/leaderboard.js]: Still added the onClick handler, but in the parent component.

**Comment by kshahzada** [comment]
Seems pretty sketch... but I'll accept it

**Pull request merged**